examples/mnist/spark/mnist_dist.py

```diff
@@ -98,27 +98,22 @@ def feed_dict(batch):
       accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32), name="accuracy")
       tf.summary.scalar("acc", accuracy)
 
-      saver = tf.train.Saver()
       summary_op = tf.summary.merge_all()
-      init_op = tf.global_variables_initializer()
 
-    # Create a "supervisor", which oversees the training process and stores model state into HDFS
+    # Create a "MonitoredTrainingSession", which oversees the training process and stores model state into HDFS
     logdir = ctx.absolute_path(args.model)
     print("tensorflow model path: {0}".format(logdir))
-
+    hooks = [tf.train.StopAtStepHook(last_step=100000)]
+    
     if job_name == "worker" and task_index == 0:
       summary_writer = tf.summary.FileWriter(logdir, graph=tf.get_default_graph())
 
     if args.mode == "train":
-      sv = tf.train.Supervisor(is_chief=(task_index == 0),
-                               logdir=logdir,
-                               init_op=init_op,
-                               summary_op=None,
-                               summary_writer=None,
-                               saver=saver,
-                               global_step=global_step,
-                               stop_grace_secs=300,
-                               save_model_secs=10)
+      with tf.train.MonitoredTrainingSession(master=server.target,
+                                             is_chief=(task_index == 0),
+                                             checkpoint_dir=logdir,
+                                             hooks=hooks,
+      ) as mon_sess:
```

Comment:
This code throws an error for me (when trying the [Standalone instructions](https://github.com/yahoo/TensorFlowOnSpark/wiki/GetStarted_Standalone)).  Also, this wouldn't work for `args.mode == "inference"`.  Can you update accordingly?

Reply:
Can you show me the error？

Reply:
```Traceback (most recent call last):
  File "/Users/leewyang/devpub/TensorFlowOnSpark/examples/mnist/spark/mnist_spark.py", line 18, in <module>
    import mnist_dist
  File "/Users/leewyang/devpub/TensorFlowOnSpark/examples/mnist/spark/mnist_dist.py", line 117
    else:
    ^
IndentationError: expected an indented block
```

Reply:
I'am coming! I have fix some problems and I can run successfully on Hadoop Cluster.